core/src/main/java/uk/ac/bbsrc/tgac/miso/core/util/PaginationFilter.java

```diff
@@ -58,14 +58,15 @@ public static PaginationFilter boxUse(long id) {
     };
   }
 
-  public static <M extends Identifiable, D> List<D> bulkSearch(Collection<String> names, PaginatedDataSource<M> service,
-      Function<M, D> dto) {
+  public static <M extends Identifiable, D, X extends RuntimeException> List<D> bulkSearch(Collection<String> names,
+      PaginatedDataSource<M> service,
+      Function<M, D> dto, Function<String, X> makeException) {
     return names.stream()//
         .filter(name -> !LimsUtils.isStringBlankOrNull(name))//
         .flatMap(WhineyFunction.flatRethrow(name -> {
           Collection<M> matches = service.list(0, 0, true, "id", exactQuery(name));
           if (matches.isEmpty()) {
-            throw new IllegalArgumentException(String.format("Cannot find \"%s\".", name));
+            throw makeException.apply(String.format("Cannot find \"%s\".", name));
```

Comment:
Should use the `RestException(String, Status)` constructor instead to provide `Status.BAD_REQUEST` since this is a client error and RestException will otherwise default to internal server error